qemu/tests/cfg/virtio_mode.cfg

```diff
@@ -30,12 +30,12 @@
                 driver_verifier += " ndis"
             device_type = "virtio-net-pci"
             device_name = "Red Hat VirtIO Ethernet Adapter.*"
-            pci_id_pattern = "(\d+:\d+\.\d+)\s+Ethernet controller:.*?Virtio network device"
+            pci_id_patterns = "(\d+:\d+\.\d+).*?1af4:1000 (\d+:\d+\.\d+).*?1af4:1041"
```

Comment:
I have not check the details about the id, but to support multiple ids in the pattern, it can be:
`"(\d+:\d+\.\d+).*?1af4:(1000|1041)"`

Reply:
@yanan-fu applying your proposal, not only the bus is obtained but the pci ID too, leading the case to fail

```bash
[stdlog] 2023-07-17 05:20:38,145 aexpect.client DEBUG| [10.73.212.125] Sending command: lspci -n
[stdlog] 2023-07-17 05:20:38,306 avocado.test DEBUG| the pci_n value: ('05:00.0', '1041')
...
[stdlog] 2023-07-17 05:20:38,309 avocado.test ERROR|   File "/root/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/virtio_mode.py", line 109, in verify_virtio_mode_guest_linux
[stdlog] 2023-07-17 05:20:38,309 avocado.test ERROR|     cmd = "grep . /sys/bus/pci/devices/0000:%s/virtio*/features" % pci_n
[stdlog] 2023-07-17 05:20:38,309 avocado.test ERROR| TypeError: not all arguments converted during string formatting
```

Reply:
`(?:1000|1041)`
You had to add `?:` to make it a non-capturing group, see: https://docs.python.org/3/library/re.html#regular-expression-syntax

Reply:
With the new change, `pci_n` is not the initial thing any more.
You can choose:
1. use the way @PaulYuuu  mentioned above, and keep using `pci_n = re.findall(pci_id_pattern, pci_info)`
2. use `pci_n = re.search(pci_id_pattern, pci_info).group(1)`

Reply:
Applied @PaulYuuu suggestion, all the cases passed 
```bash
 (01/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_netkvm.with_modern.q35: STARTED
 (01/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_netkvm.with_modern.q35: PASS (53.39 s)
 (02/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_netkvm.with_transitional.q35: STARTED
 (02/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_netkvm.with_transitional.q35: PASS (53.70 s)
 (03/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_netkvm.with_legacy.q35: STARTED
 (03/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_netkvm.with_legacy.q35: PASS (53.57 s)
 (04/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_builtin.with_modern.q35: STARTED
 (04/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_builtin.with_modern.q35: PASS (54.64 s)
 (05/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_builtin.with_transitional.q35: STARTED
 (05/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_builtin.with_transitional.q35: PASS (54.85 s)
 (06/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_builtin.with_legacy.q35: STARTED
 (06/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_builtin.with_legacy.q35: PASS (54.08 s)
 (07/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_random.with_modern.q35: STARTED
 (07/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_random.with_modern.q35: PASS (54.78 s)
 (08/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_random.with_transitional.q35: STARTED
 (08/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_random.with_transitional.q35: PASS (56.13 s)
 (09/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_random.with_legacy.q35: STARTED
 (09/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_viorng.with_rng_random.with_legacy.q35: PASS (53.84 s)
 (10/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_vioscsi.with_modern.q35: STARTED
 (10/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_vioscsi.with_modern.q35: PASS (51.66 s)
 (11/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_vioscsi.with_transitional.q35: STARTED
 (11/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_vioscsi.with_transitional.q35: PASS (54.38 s)
 (12/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_vioscsi.with_legacy.q35: STARTED
 (12/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_vioscsi.with_legacy.q35: PASS (54.17 s)
 (13/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_balloon.with_modern.q35: STARTED
 (13/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_balloon.with_modern.q35: PASS (53.54 s)
 (14/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_balloon.with_transitional.q35: STARTED
 (14/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_balloon.with_transitional.q35: PASS (54.79 s)
 (15/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_balloon.with_legacy.q35: STARTED
 (15/15) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.virtio_mode.with_balloon.with_legacy.q35: PASS (54.35 s)
RESULTS    : PASS 15 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```